src/services/aof.rs

```diff
@@ -0,0 +1,130 @@
+use anyhow::Result;
+use bytes::{BufMut, Bytes, BytesMut};
+
+/// Trait for an Append-Only File (AOF) abstraction.
+
+pub trait TAof {
+    /// Appends a single `AofEntry` to the log.
+    fn append(
+        &mut self,
+        op: &WriteOperation,
+    ) -> impl std::future::Future<Output = Result<()>> + Send;
+
+    /// Replays all logged entries from the beginning of the AOF, calling the provided callback `f`
+    /// for each entry.
+    ///
+    /// The callback `f(AofEntry)` receives each entry in the order it was appended.
+    fn replay<F>(&mut self, f: F) -> impl std::future::Future<Output = Result<()>> + Send
+    where
+        F: FnMut(AofEntry) + Send;
+
+    /// Forces pending writes to be physically recorded on disk.
+    fn fsync(&mut self) -> impl std::future::Future<Output = Result<()>> + Send;
+
+    /// Returns the highest offset **successfully** appended or replayed so far.
+    ///
+    /// - If zero, means no entries were successfully written.
+    /// - This offset should never skip a number if the implementation fails (e.g., fails mid-write).
+    fn latest_offset(&self) -> u64;
+}
+
+/// A single entry in the Append-Only File (AOF).
+#[derive(Debug, Clone, PartialEq, Eq)]
```

Comment:
Instead of having this thin wrapper around write operation that'd be extracted from user request, perhaps we could add `TAof` aspect to existing implementation? WDYT?

Reply:
I do agree there is no benefit in its current form. The `AofEntry` was not a thin wrapper when I first wrote it. It included additional metadata like offset, timestamp, and CRC. I've removed those to keep it simple for now since I haven't fully fleshed out how I want to represent a fully qualified log record. I'll remove the AofEntry and use the WriteOperation directly here. We can re-visit if we decide to embed extra fields to the entry.